### PR TITLE
the service worker

### PR DESCRIPTION
**Pull request by prkhrv**
just enter the url you are hosting on

### CODE REVIEW

**Comment by jr-cologne** [comment]
Hi @prkhrv,

first of all, thanks for your contribution.
Please give me some time to test and review your changes.

I will get back to you next weekend at the latest.
I didn't expected contributions flying in so quickly. 😄 

Best regards,
@jr-cologne

**Comment by prkhrv** [comment]
Ok, You can ask me if you get stuck somewhere with the service worker.

**Comment by jr-cologne** [comment]
Sounds great! :+1: 

Have a nice day!

File: serviceWorker.js [hunk 1]
```diff
@@ -0,0 +1,214 @@
+//Cache polyfil to support cacheAPI in all browsers
+importScripts('./cache-polyfill.js');
```
Comment (jr-cologne) [inline serviceWorker.js]: What is this and do I even need this? I don't think so.

File: serviceWorker.js [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+var cacheName = 'cache-v4';
```
Comment (jr-cologne) [inline serviceWorker.js]: I don't quite understand what naming the cache is good for. Is this just going to stay consistent?

File: serviceWorker.js [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+//Files to save in cache
+var files = [
```
Comment (jr-cologne) [inline serviceWorker.js]: Explicitly declaring these files is no real option in my opinion. Is there another way?

File: serviceWorker.js [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+      console.info("Old caches are cleared!");
+      // To tell the service worker to activate current one
+      // instead of waiting for the old one to finish.
+      return self.clients.claim();
+    })
+  );
+});
+
+/*
+  PUSH EVENT: triggered everytime, when a push notification is received.
```
Comment (jr-cologne) [inline serviceWorker.js]: I don't want push notifications, so this can be removed.

File: serviceWorker.js [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+      { 'action': 'yes', 'title': 'I ♥ this app!'},
+      { 'action': 'no', 'title': 'I don\'t like this app'}
+    ]
+  };
+
+  event.waitUntil(self.registration.showNotification(title, body));
+});
+
+/*
+  BACKGROUND SYNC EVENT: triggers after `bg sync` registration and page has network connection.
```
Comment (jr-cologne) [inline serviceWorker.js]: Do not want that as well. Please remove.

File: serviceWorker.js [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+      })
+      .catch((error) => {
+        console.error(error);
+      })
+    );
+  }
+});
+
+/*
+  NOTIFICATION EVENT: triggered when user click the notification.
```
Comment (jr-cologne) [inline serviceWorker.js]: The same goes for here.

File: serviceWorker.js (hunk 1 above)
Comment (prkhrv) [reply serviceWorker.js]: Yeah it's actually needed to make your PWA to work offline. I will add this file.

File: serviceWorker.js (hunk 2 above)
Comment (prkhrv) [reply serviceWorker.js]: Well Its for offline , workflow of the app!

File: serviceWorker.js (hunk 3 above)
Comment (prkhrv) [reply serviceWorker.js]: I have always used it like this, I will add these files soon btw. And also I will ask around if there is any other way.

File: serviceWorker.js (hunk 4 above)
Comment (prkhrv) [reply serviceWorker.js]: Ok :-)

File: serviceWorker.js (hunk 6 above)
Comment (prkhrv) [reply serviceWorker.js]: Will change them all

File: serviceWorker.js (hunk 1 above)
Comment (jr-cologne) [reply serviceWorker.js]: But this is just a polyfill, isn't it? In modern browsers, it should work without it.

File: serviceWorker.js (hunk 6 above)
Comment (jr-cologne) [reply serviceWorker.js]: Great, thank you. :)

File: serviceWorker.js (hunk 1 above)
Comment (prkhrv) [reply serviceWorker.js]: I guess You are Correct! I just keep this, you know? for redundancy xD

Comment (prkhrv) [reply serviceWorker.js]: btw, https://prkhrv.github.io  check out this PWA version of Flappy bird I made :)

Comment (jr-cologne) [reply serviceWorker.js]: Best approach! 😄 

We will just try it out and can remove that later.

File: serviceWorker.js (hunk 3 above)
Comment (jr-cologne) [reply serviceWorker.js]: Ok, no problem. Just to get this started, we could actually stick to that.

Later, it should definitely do this automatically.

**Comment by prkhrv** [comment]
Hey, I have added the JS files

**Comment by jr-cologne** [comment]
Hi @prkhrv,
I really appreciate that you are trying to help me.
However, this is going a bit off. You are essentially adding the code from your own repository which does not really fit into my website.
Reusing code is totally fine, but then the code should actually be reusable.
Simply adding the same code is not effective and helpful at all since I would still need to rewrite everything.
I cannot expect you to deliver the perfect code which can be merged in without any changes, but it should at least have more value than me just copying random code from the Internet.
What do you think? Please do not feel offended. Just want to be honest with you.
Best regards,
@jr-cologne

**Comment by prkhrv** [comment]
@jr-cologne Well I see the point you are saying but I guess, these files should work perfect, I made them while working on my First PWA, I have reused them since then. It has always worked fine, Though You have to make some changes of course. Which I will do if you are facing any problems.

Best regards,
@prkhrv

**Comment by jr-cologne** [comment]
@prkhrv I am not saying that your code does not work (I will test it), but you are basically adding various useless files on top of that like app.js, menu.js, and so on.

**Comment by prkhrv** [comment]
actually app.js, menu.js are for redundancy for Android and IOS. they are required for the manifest to work properly, You can remove these files if not needed.

**Comment by jr-cologne** [comment]
Hi @prkhrv,

Sorry for the late reply. I was on vacation.

I think I am going to close this PR. I really appreciate the efforts you put into this project trying to help me, but your code just doesn't fit so well into this project as I already pointed out.

Like I have promised you, I tested your code and it generally sort of worked beside some flaws. However, I would need to spend a relatively huge amount of time just clearing out unneeded stuff as well as adjusting things to my own needs in order to preserve a clean codebase.

I am definitely going to work on a Service Worker for my website, but then completely from the ground up which just feels a lot cleaner and better to me. Of course, you are invited to contribute to this process.
Later, I am going to create a new branch called `pwa` as well as a corresponding PR for this.
Then, we could together work on the Service Worker, but only with a better approach talking about changes beforehand to prevent any kind of wasted work, if you know what I mean.

I really hope you are fine with my decision. Again, thanks a lot so far for your help.

Best regards,
@jr-cologne

**Pull request closed without merging**